internal/ingress/controller/template/template.go

```diff
@@ -161,6 +162,20 @@ var (
 	}
 )
 
+// escapeLocationPathVar will replace the $ character with ${literal_dollar}
+// which is made to work via the following configuration in the http section of
+// the template:
+// geo $literal_dollar {
+//     default "$";
+// }
+func escapeLocationPathVar(input interface{}) string {
```

Comment:
This is not necessarily about location path var and can be useful for other variables that might need to include `$` in the value. What about `escapeLiteralDollar`?